ug: Add user guides on custom CI for OS and apps (#461)
foundriesio/docs: source/user-guide/custom-ci/custom-ci-for-rootfs.rst

.. _ug-custom-ci-for-rootfs:

> Custom CI For RootFS

Reviewer: We might want to think about this "RootFS" term. I'm not sure we've used this term before. I can see us explaining this to people a lot by saying something like "You know, a Platform build". So I wonder if we should call it something like that Yocto/OE/Platform instead.

Reviewer 2: I was able to assume the correct context by the title. but that's likely my familiarity.
My recommendation would be to update `custom-ci.rst` to have a description of each child page, and what it will be discussing similar to the [FoundriesFactory](https://docs.foundries.io/latest/reference-manual/factory/factory.html) section.
This is actually something I want to implement throughout our docs. However it is likely best to handle that separately.

Author: >My recommendation would be to update custom-ci.rst to have a description of each child page, and what it will be discussing similar to the [FoundriesFactory](https://docs.foundries.io/latest/reference-manual/factory/factory.html) section.

I updated `custom-ci` with "what it will be discussing".

.. prompt:: text

    IMAGE_FSTYPES:remove = "ostreepush garagesign garagecheck"

2. Run the following to bitbake both a system image (if flashing of a device is needed) and the ostree repo.

.. prompt:: text

    bitbake lmp-factory-image

3. Run the following to bitbake just the ostree repo.

.. prompt:: text

    bitbake lmp-factory-image -c do_image_ostreecommit

As a result, you should get an ostree repo that contains a rootfs that can be delivered to your devices via the OTA service. For example:

.. prompt:: text

    ./deploy/images/intel-corei7-64/ostree_repo

Push OSTree Repo To Cloud
-------------------------
The :ref:`ref-linux-dev-container` includes utilities called fiopush and fiocheck that pushes an ostree repo to the foundries multi-tenant storage based on GCS.
You need an auth token to run these commands.
The token can be obtained at `FoundriesFactory WebApp`_, and it should have ``targets:read-update`` scope.
Run ``fiopush -factory <factory> -repo ./deploy/images/intel-corei7-64/ostree_repo -token <fio-token>`` to push the bitbaked ostree repo to the FoundriesFactory storage.

Add OSTree Target

> -----------------

Reviewer 2: ```suggestion
-----------------

```

Once the ostree repo carrying rootfs is built and pushed to the cloud, a user can add a new Target referring to it.

The rootfs committed to the ostree repo is referred by the commit hash.
Run ``find ./deploy/images/intel-corei7-64/ostree_repo -name *.commit`` or ``ostree --repo ./deploy/images/intel-corei7-64/ostree_repo rev-parse <machine|hardware ID>``
to obtain the hash.

> Then, you can run the ``fioctl targets add`` add new Target that refer to the given ostree-based rootfs, e.g.

Reviewer 2: ```suggestion
Then, you can run ``fioctl targets add`` to add a new Target that references the given ostree-based rootfs, e.g.,
```

.. prompt:: text

    fioctl targets add --type ostree --tags master,devel --src-tag master intel-corei7-64 094a6d77b7053f2fec1e5e4ccd83c38cb89174f644303c6bb09693648be98912

> Check Created OSTree Target

Reviewer 2: ```suggestion
Check the OSTree Target
```

> ---------------------------

Reviewer 2: ```suggestion
-----------------------

```

> Use ``fioctl targets list`` and ``fioctl targets show`` commands to check whether the new Target is registered in the Foundries OTA service

Reviewer 2: ```suggestion
Use ``fioctl targets list`` and ``fioctl targets show`` commands to check whether the new Target is registered in the OTA service
```

and whether their content is correct.

If ``aktualizr-lite`` is configured for one of the new Target's tags, then it should be able to enlist and install the Target.

.. prompt:: text

    aktualizr-lite list
    ...
    info: 1589	sha256:094a6d77b7053f2fec1e5e4ccd83c38cb89174f644303c6bb09693648be98912

Then during the update one can see the log saying that aklite is downloading the expected ostree commit.

.. prompt:: text

    info: Fetching ostree commit 094a6d77b7053f2fec1e5e4ccd83c38cb89174f644303c6bb09693648be98912 from https://storage.googleapis.com/ota-lite-ostree-eu/094a6d77b7053f2fec1e5e4ccd83c38cb89174f644303c6bb09693648be98912
    ...
    aktualizr-lite status
    info: Active image is: 1589	sha256:00b2ad4a1dd7fe1e856a6d607ed492c354a423be22a44bad644092bb275e12fa

> 

Reviewer 2: ```suggestion
```
